### ISSUE TITLE
[UX] Rich Text Editor: Image & image caption improvement suggestions.

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#1093]
![backdrop-ckeditor-add_image_with_caption](https://cloud.githubusercontent.com/assets/2423362/9080991/0f06a900-3b9c-11e5-924e-7d82d8cfd1f5.png)

A few issues I see here...

1. Can we do something to loose the vertical scrollbar (this is on latest Firefox x64)?
2. Why the "add caption" checkbox does not have a text field to allow entering the caption directly?
3. I would like to see a "Use the alternative text as caption" checkbox there, so that people that have already entered alt text to their images would only have to edit the image -> click "enable caption" -> click "same as alt text" -> save, done.

**Comment by quicksketch** [comment]
> Can we do something to loose the vertical scrollbar (this is on latest Firefox x64)?

I fiddled with this for a while. There's something funny with the hidden buttons at the bottom of the form that they're still taking up unnecessary space.

> Why the "add caption" checkbox does not have a text field to allow entering the caption directly?

This is primarily because captions may contain HTML, so once the caption is added you can use the bold/link/etc buttons. However, in general I completely agree. Most captions are just plain text and I'd expect to be able to type the caption in the dialog as well.

> I would like to see a "Use the alternative text as caption" checkbox there, so that people that have already entered alt text to their images would only have to edit the image -> click "enable caption" -> click "same as alt text" -> save, done.

I don't think this would be appropriate. Captions are not the same as alternative text.
https://wordpress.org/support/topic/caption-and-alt-text-should-not-be-the-same
https://en.wikipedia.org/wiki/Wikipedia:Alternative_text_for_images#Examples

However, in situations where you wanted to use the same for both, a simple copy/paste is going to be simple enough for most users. I just don't think that we should be leading users to believe that alt text and captions are the same thing.

**Comment by klonos** [comment]
Yeah, fair enough about the caption alt text thing. I was just thinking similar forms where for example there is a "same as home address" checkbox for the shipping address so that users don't have to retype things.

**Comment by BWPanda** [comment]
> This is primarily because captions may contain HTML...

I've just noticed that URLs in captions don't automatically convert to links. It seems the filters aren't being run on them maybe...? Be nice to see URLs in captions automatically become links.

**Comment by klonos** [comment]
I propose the following:

1. Replace the "Add a caption" checkbox with a "Caption" field. If the user adds caption text, then we add a caption when the user clicks on Save/Update and include that text in it. If the user leaves the caption field blank, we do not add a caption. Compare the UX before/after:

before: add image -> do not check the "add a caption" checkbox -> save
after: add image -> do not add text in the caption field -> save

...so basically same number of clicks

before: add/edit image -> check the "add a caption" checkbox -> save -> add caption
after: add/edit image -> add text in the caption field -> save

...1 less click = UX+

PS: @BWPanda I have moved your suggestion to #3383, since that one sounds like a bug.
